src/metabase/integrations/common.clj

```diff
@@ -9,29 +10,31 @@
             [toucan.db :as db]))
 
 (defn sync-group-memberships!
-  "Update the PermissionsGroups a User belongs to, adding or deleting membership entries as needed so that Users is only
-  in `new-groups-or-ids`. Ignores special groups like `all-users`."
-  [user-or-id new-groups-or-ids sync-admin-group?]
-  ;; if you AREN'T syncing the admin group, it's a "special group", which means it's excluded from being added
-  ;; or removed from a user
-  (let [special-group-ids  (if (false? sync-admin-group?)
-                             #{(u/the-id (group/admin)) (u/the-id (group/all-users))}
-                             #{(u/the-id (group/all-users))})
+  "Update the PermissionsGroups a User belongs to, adding or deleting membership entries as needed so that Users is
+  only in `new-groups-or-ids`. Ignores special groups like `all-users`, and only touches groups with mappings set."
+  [user-or-id new-groups-or-ids mapped-groups-or-ids sync-admin-group?]
+  (let [included-group-ids (cond-> (set (map u/the-id mapped-groups-or-ids))
+                             sync-admin-group? (conj (u/the-id (group/admin))))
+        excluded-group-ids (cond-> #{(u/the-id (group/all-users))}
+                             (not sync-admin-group?) (conj (u/the-id (group/admin))))
         user-id            (u/the-id user-or-id)
-        ;; Get a set of Group IDs the user currently belongs to
+        ;; Get a set of mapped Group IDs the user currently belongs to
         current-group-ids  (db/select-field :group_id PermissionsGroupMembership
                                             :user_id  user-id
-                                            :group_id [:not-in special-group-ids])
-        new-group-ids      (set (map u/the-id new-groups-or-ids))
-        ;; determine what's different between current groups and new groups
+                                            ;; Add nil to included group ids to ensure valid SQL if set is empty
+                                            :group_id [:in (conj included-group-ids nil)]
```

Comment:
This was a workaround for a SQL error when `included-group-ids` is an empty set, since it seems to be converted to `IN NULL`. Not sure if there's a more elegant way to solve this.

Reply:
If `included-group-ids` is empty, just don't run the query in the first place. Do 

```clj
(when (seq included-group-ids)
   ...)
```